diff --git a/runtime/sak.c b/runtime/sak.c
index 7499dbf..6831075 100644
--- a/runtime/sak.c
+++ b/runtime/sak.c
@@ -74,9 +74,7 @@ void encode_C_literal(char_os *path)
 {
   char_os c;
 
-#ifdef _WIN32
   putchar('L');
-#endif
   putchar('"');
 
   while ((c = *path++) != 0) {
@@ -87,11 +85,6 @@ void encode_C_literal(char_os *path)
       printf("\\\"");
     } else if (c == '\n') {
       printf("\\n");
-#ifndef _WIN32
-    /* On Unix, nothing else needs escaping */
-    } else {
-      putchar(c);
-#else
     /* On Windows, allow 7-bit printable characters to be displayed literally
        and escape everything else (using the older \x notation for increased
        compatibility, rather than the newer \U. */
@@ -99,7 +92,6 @@ void encode_C_literal(char_os *path)
       putwchar(c);
     } else {
       printf("\\x%04x", c);
-#endif
     }
   }
 
